asapdiscovery-data/asapdiscovery/data/tests/test_ligand_schema_v2.py

```diff
@@ -141,41 +182,50 @@ def test_ligand_sdf_rountrip_data_only(
 
 def test_ligand_oemol_rountrip(moonshot_sdf):
     mol = load_openeye_sdf(str(moonshot_sdf))
-    l1 = Ligand.from_oemol(mol)
+    l1 = Ligand.from_oemol(mol, compound_name="blahblah")
     mol_res = l1.to_oemol()
-    l2 = Ligand.from_oemol(mol_res)
+    l2 = Ligand.from_oemol(mol_res, compound_name="blahblah")
     assert l2 == l1
 
 
 def test_ligand_oemol_rountrip_data_only(moonshot_sdf):
     mol = load_openeye_sdf(str(moonshot_sdf))
     l1 = Ligand.from_oemol(mol, compound_name="blahblah")
     mol_res = l1.to_oemol()
-    l2 = Ligand.from_oemol(mol_res)
+    l2 = Ligand.from_oemol(mol_res, compound_name="blahblah")
     assert l1.data_equal(l2)
 
 
 def test_get_set_sd_data(moonshot_sdf):
-    l1 = Ligand.from_sdf(moonshot_sdf)
+    l1 = Ligand.from_sdf(moonshot_sdf, compound_name="blahblah")
     data = {"test_key": "test_value", "test_key2": "test_value2", "test_key3": "3"}
     l1.set_SD_data(data)
     data_pulled = l1.get_SD_data()
     assert data_pulled == data
 
 
 def test_print_sd_data(moonshot_sdf):
-    l1 = Ligand.from_sdf(moonshot_sdf)
+    l1 = Ligand.from_sdf(moonshot_sdf, compound_name="blahblah")
     data = {"test_key": "test_value", "test_key2": "test_value2", "test_key3": "3"}
     l1.set_SD_data(data)
     l1.print_SD_data()
 
 
+def test_clear_sd_data(moonshot_sdf):
+    l1 = Ligand.from_sdf(moonshot_sdf, compound_name="blahblah")
+    data = {"test_key": "test_value", "test_key2": "test_value2", "test_key3": "3"}
+    l1.set_SD_data(data)
+    l1.clear_SD_data()
+    assert l1.get_SD_data() == {}
+
+
+@pytest.mark.parametrize("tags", [{"test_key": "test_value"}, {}])
 @pytest.mark.parametrize("exp_data_vals", [{"pIC50": 5.0}, {}])
 @pytest.mark.parametrize("moonshot_compound_id", ["test_moonshot_compound_id", None])
 @pytest.mark.parametrize("manifold_vc_id", ["ASAP-VC-1234", None])
 @pytest.mark.parametrize("manifold_api_id", [uuid4(), None])
 @pytest.mark.parametrize("compchem_id", [uuid4(), None])
-@pytest.mark.parametrize("compound_name", ["test_name", None])
+@pytest.mark.parametrize("compound_name", ["test_name"])
 def test_ligand_sdf_rountrip_SD(
```

Comment:
can we also add a test to make sure we can check if all of the data of two ligands is the same / different? I think using `flush_attrs_to_SD_data`

Reply:
I have added a `full_equal` method to check for every-field equality.

Reply:
nice, thanks!